### PR TITLE
Display license status in management console

### PR DESCRIPTION
**Pull request by attfarhan**
Fixes https://github.com/sourcegraph/sourcegraph/issues/1432.

As discussed in the issue, there is a lot of copy-pasta. Primarily from the license code from `enterprise/cmd/frontend/licensing/` into the new enterprise management console binary, and several components from `web/src/enterprise` to display the product subscription. Please leave comments if any questions. I would recommend actually running this locally to test too.

### CODE REVIEW

File: cmd/management-console/shared/main.go [hunk 1]
```diff
@@ -0,0 +1,345 @@
+// Command management-console provides a failsafe editor for the critical
+// configuration options for the Sourcegraph instance.
+//
+// 🚨 SECURITY: No authentication is done by the management console.
+// It is currently the user's responsibility to:
+//
+// 1. Limit access to the management console by not exposing its port.
+// 2. Ensure that the management console's responses are never propagated to
+//    unprivileged users.
+//
```
Comment (slimsag) [inline cmd/management-console/shared/main.go]: This whole security line is actually outdated, I forgot to update it. We do in fact do authentication and exposing the port is OK. Let's update package description to just keep the first two lines.

File: cmd/management-console/shared/license.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
+package shared
```
Comment (slimsag) [inline cmd/management-console/shared/license.go]: Is this entire file copy-pasta? Can we just move it out of `enterprise/cmd/frontend/licensing/` and into `enterprise/pkg/license/licenseutil` ? I mostly meant in the issue that copy-pasta was fine in cases where it would be really hard to share (e.g. would require us to refactor large amounts of frontend code and publish an npm package or something)

File: enterprise/cmd/management-console/licensing.go [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+	return "Sourcegraph Enterprise" + name
+}
+
+var noLicenseMaximumAllowedUserCount uint = 200
+
+// Make the Site.productSubscription GraphQL field return the actual info about the product license,
+// if any.
+func init() {
+	fmt.Println("INIT")
```
Comment (slimsag) [inline enterprise/cmd/management-console/licensing.go]: rm

**Comment by slimsag** [comment]
@attfarhan I think we will need to do more work to land this and be confident it works well, so I don't think we can land this in 3.0 safely. In specific:

- This PR introduces too much copied code (I anticipated _some_ due to difficulty in sharing frontend code -- but I didn't realize how hard reusing our licensing logic would be, this isn't your fault).
- After spending an hour or so trying to find a more proper way to separate out the licensing logic so that we can share the code nicely, I realize the code actually relies on both redis and postgres access. This probably worked fine in the dev server, but this means we need to configure the management console to have access to redis in deployments in general and I am not sure if that is a good choice yet or if we can remove that dependency.
- For cleanliness, we shouldn't have much licensing code in `cmd/management-console/shared/main.go` and need to move e.g. `serveLicense` out of there and into `enterprise/cmd/management-console` but this may introduce other code structure challenges.
- I am thinking we need to find a good way to share the `ProductSubscriptionStatus` component given it has some pretty hefty logic and fixing this after merging / later on would be difficult to do.

Basically, this is a lot more complex than I originally thought it was going to be (my fault, though I suspect you also noticed this while working on it), and I think we need to spend more time here before we can ship this confidently.

Let's instead aim to land this in 3.1, I will find time soon to look into the above Go license code refactoring issues and maybe you can investigate sharing `ProductSubscriptionStatus` code more.

**Comment by felixfbecker** [comment]
> and several components from web/src/enterprise to display the product subscription.

Why do these have to be copied instead of moved to the open source codebase?

**Comment by slimsag** [comment]
Closing for now, we can revisit when important to prioritize again.
